### PR TITLE
Fix failing API tests

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?

Fixed failing API tests by throwing a relevant error message when `uid` is undefined in `buildRelationsStore`

### CODE REVIEW

File: packages/core/strapi/lib/services/entity-validator/index.js [hunk 1]
```diff
@@ -247,9 +247,14 @@ const createValidateEntity =
  * @returns {Object}
  */
 const buildRelationsStore = ({ uid, data }) => {
+  if (!uid) {
+    throw new ValidationError(`cannot build relations store: "uid" is undefined`);
```
Comment (alexandrebodin) [inline packages/core/strapi/lib/services/entity-validator/index.js:251]:
```suggestion
    throw new ValidationError(`Cannot build relations store: "uid" is undefined`);
```

Comment (alexandrebodin) [inline packages/core/strapi/lib/services/entity-validator/index.js:251]: I'm not sure this error message will be helpful for a user. if it is realted to missing __component in the DZ then I would recommend catching this earlier to add better errors. (We can do that in another PR)

**Review by alexandrebodin — CHANGES_REQUESTED** [review]

File: packages/core/strapi/lib/services/entity-validator/index.js (hunk 1 above)
Comment (nathan-pichon) [reply packages/core/strapi/lib/services/entity-validator/index.js:251]: I would suggest to have meaningful error message at every level of our code (maybe until we migrate to TS ? As TS build would fail if some mandatory parameter aren't provided).
Here we can maybe just throw `A model uid is required to build relations` ?

Comment (alexandrebodin) [reply packages/core/strapi/lib/services/entity-validator/index.js:251]: I wasn't clear enough. We should keep this error but add another one earlier that is more context based with DZ. I don't think we have to wait for TS to do this.

Comment (nathan-pichon) [reply packages/core/strapi/lib/services/entity-validator/index.js]: For sure ! More meaningful errors 🎉

**Review by alexandrebodin — APPROVED** [review]

**Review by nathan-pichon — APPROVED** [review]

**Review by Marc-Roig — APPROVED** [review]
